### PR TITLE
Upgrade pip in venv and use --no-cache

### PR DESCRIPTION
**Pull request by bac**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/62/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/63/
Test PASSed.

**Comment by jrwren** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/64/
Test PASSed.

File: tests/00-setup [hunk 1]
```diff
@@ -36,9 +36,10 @@ createvenv() {
         [ $updated_downloadcache -eq 0 ] && bzr up $DOWNLOADCACHE
         virtualenv --distribute --extra-search-dir=$DOWNLOADCACHE $VENV
         . $ACTIVATE && \
-            yes w | env BZR_PLUGIN_PATH='-user' \
-            pip install --find-links releases --find-links jujugui-deps jujugui
-            pip install -r $TEST_REQUIREMENTS --find-links releases --find-links deps --find-links $DOWNLOADCACHE --no-allow-external --no-index
+        pip install --upgrade pip && \
```
Comment (jaycee) [inline tests/00-setup]: Do we really want to upgrade pip everytime? What's the logic here?

**Comment by jaycee** [comment]
@bac this is :+1: for me assuming a good explanation of the "always upgrade pip" thing. I'm confused by that.

File: tests/00-setup (hunk 1 above)
Comment (bac) [inline tests/00-setup]: It is the pip in the venv not the system one.

The intent is to get a modern pip on trusty.  I guess I could only do it if the current is < 7.  

I'll see what that looks like.

Comment (jaycee) [inline tests/00-setup]: I was aware it was the venv one; it didn't occur to me that the venv one wouldn't be the latest. I'm fine with this as is then, thanks.

**Comment by jaycee** [comment]
:+1: confirmed, thanks @bac.

**Comment by bac** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-charm-merge

**Pull request merged**